Mathlib/Algebra/GroupWithZero/Hom.lean

```diff
@@ -214,6 +214,26 @@ instance {β} [CommMonoidWithZero β] : Mul (α →*₀ β) where
     { (f * g : α →* β) with
       map_zero' := by dsimp; rw [map_zero, zero_mul] }
 
+/-- The trivial homomorphism between monoids with zero, sending everything to 1 other than 0. -/
+noncomputable def trivial (M N : Type*) [MulZeroOneClass M] [MulZeroOneClass N]
+    [DecidablePred fun x : M ↦ x = 0] [Nontrivial M] [NoZeroDivisors M] :
+    M →*₀ N where
+  toFun x := if x = 0 then 0 else 1
```

Comment:
This is a slightly funny name. Shouldn't this be `1`, since `trivial * f = f * trivial = f` for all `f : M →*₀ N`?

Reply:
It felt icky naming it 1 when 1 in other Pi contexts often means either send _everything_ to 1, or Pi.single 1.

Reply:
Done, but `trivial * f = f * trivial = f` for all `f : M →*₀ N?` is not true. This `one` is absorbing.